src/ig/IGExporter.ts

```diff
@@ -683,13 +746,227 @@ export class IGExporter {
+      'lists those files generated from project configs or the contents in the ig-data folder.',
+      '',
+      ''
+    ].join(EOL);
+
+    let rows: string[][] = [];
+    this.outputLog.forEach((details, outputPath) => {
+      // Make output paths relative to output
+      const output = path.relative(igPath, outputPath);
+      // Make input paths relative to output (except { placeholder } inputs)
+      const inputs = details.inputs
+        .map(input => (/^\{.+\}$/.test(input) ? input : path.relative(igPath, input)))
+        .sort()
+        .join(', ');
+      rows.push([output, details.action, inputs]);
+    });
+
+    rows = [
+      ['SUSHI-GENERATED FILE', 'ACTION', 'INPUT FILE(S)'],
+      ...orderBy(rows, ['1', '0'], ['desc', 'asc'])
+    ];
+
+    outputFileSync(path.join(igPath, 'SUSHI-GENERATED-FILES.md'), `${intro}${table(rows)}`, 'utf8');
+  }
+
+  /**
+   * Recursively copies one path to another, logging the output of each file so it can be reported in the
+   * SUSHI-GENERATED-FILES.md file.
+   *
+   * @param inputPath {string} - the input path to copy
+   * @param ouputPath {string} - the output path to copy to
+   */
+  private copyAsIs(inputPath: string, outputPath: string): void {
+    if (!existsSync(inputPath)) {
+      return;
+    }
+
+    copySync(inputPath, outputPath);
+    this.updateOuputLogForCopiedPath(outputPath, inputPath);
+  }
+
+  /**
+   * Recursively copies input to an output location, adding a warning to the top of the each supported
+   * file indicating that it is a generated file and should not be edited directly. Only .md and .xml
+   * files will have the warning added.  All other files are copied as-is. In addition, this function
+   * will log the output file so it can be reported in the SUSHI-GENERATED-FILES.md file.
+   *
+   * @param inputPath {string} - the input path to copy
+   * @param ouputPath {string} - the output path to copy to
+   */
+  private copyWithWarningText(inputPath: string, outputPath: string): void {
+    if (!existsSync(inputPath)) {
+      return;
+    }
+
+    if (statSync(inputPath).isDirectory()) {
+      readdirSync(inputPath).forEach(child => {
+        this.copyWithWarningText(path.join(inputPath, child), path.join(outputPath, child));
+      });
+      return;
+    }
+
+    // It's a file.  Add the warning text if applicable, otherwise copy as-is
+    let prefix: string, postfix: string;
+    if (inputPath.endsWith('.xml') || inputPath.endsWith('.md')) {
+      // It's a file that will be processed by Jekyll, so go ahead and use Jekyll comments
+      [prefix, postfix] = ['{% comment %} ', ' {% endcomment %}'];
+    } else {
+      this.copyAsIs(inputPath, outputPath);
+      return;
+    }
+
+    const extra = [
+      'To change the contents of this file, edit the original source file at:',
+      inputPath.slice(inputPath.indexOf('/ig-data/') + 1)
```

Comment:
This is more of a question, but does it make sense to also have the relative path here, similar to the path that is displayed in the table in `SUSHI-GENERATED-FILES.md`?  For example, my `index.md` file said "...edit the original source file at `ig-data/input/pagecontent/index.md`" but then the table showed me that that file was at `../test/ig/fixtures/customized-ig/ig-data/input/pagecontent/index.md`. It felt like those two paths should be the same, and I do really like the relative path. I'm open to leaving it as is since I think it would require passing an additional path to use to create the relative path.